sim/firesim-lib/src/main/scala/bridges/TestFinisherBridge.scala

```diff
@@ -0,0 +1,97 @@
+
+package firesim.bridges
+
+import chisel3._
+import chisel3.util._
+import freechips.rocketchip.config.Parameters
+import midas.widgets._
+import midas.widgets.PeekPokeBridge
+
+case class CodeElaborator(isErr: Boolean, message : String)
+
+case class TestIndicatorParams(
+  nConcurrentTests: Int     = 1,
+  codeMessage: Seq[CodeElaborator]
+) {
+  def codeBits: Int    = 16
+}                 
+
+
+class TestFinisherPortIO(val params: TestIndicatorParams) extends Bundle {
+  //Error code in case program terminates unsuccessfully
+  val errCode = UInt((params.codeBits).W)
+  val valid = Bool() 
+}
+
+class TestFinisherBridgeTargetIO(params: TestIndicatorParams) extends Bundle {
+  val clock = Input(Clock())
+  val testDone = Input(new TestFinisherPortIO(params))
+}
+
+class TestFinisherBridgeHostIO(params: TestIndicatorParams)
+                              (private val targetIO : TestFinisherBridgeTargetIO = new TestFinisherBridgeTargetIO(params)) 
+                               extends Bundle with ChannelizedHostPortIO{
+  def targetClockRef = targetIO.clock 
+  //It is just inputs from the target to indicate test completion
+  //The HostPort annotation would not work with aggregates so creating individual channels.
+  val errCodeH = InputChannel(targetIO.testDone.errCode)
+  val validH = InputChannel(targetIO.testDone.valid)
+}
+
+class TestFinisherBridge(params: TestIndicatorParams)(implicit p: Parameters) extends BlackBox
+    with Bridge[TestFinisherBridgeHostIO, TestFinisherBridgeModule] {
+  val io = IO(new TestFinisherBridgeTargetIO(params))
+  val bridgeIO = new TestFinisherBridgeHostIO(params)(io)
+
+  val constructorArg = Some(params)
+
+  generateAnnotations()
+}
+
+object TestFinisherBridge {
+  def apply(clock: Clock, finisher: TestFinisherPortIO)(implicit p: Parameters): TestFinisherBridge = {
+    val tfm = Module(new TestFinisherBridge(finisher.params))
+    tfm.io.testDone := finisher
+    tfm.io.clock := clock 
+    tfm  
+  }
```

Comment:
Consider adding an apply method that takes a Seq[Bool] and a Seq[TerminationCondition] and does the encoding using a priority encoder.

Reply:
Assuming the `Seq[Bool]` as Seq of completion and priority encoder is applied on isErr of Seq[TerminationCondition]. Addressed it.